website/docs/opengl/c1/2-hello-quad.md

```diff
@@ -0,0 +1,524 @@
+
+On top of this, a window will often contain at least two of these buffers. This is known as **double-buffering** and is imperitive for VSync to work properly. One buffer is displayed, while another is rendered to. 
+
+This is what makes up the **swapchain**, which, if you've ever looked into Direct3D, you should be familliar with. OpenGL does not use the term, nor does it allow you to manage this yourself, however it is still in the background, and it is helpful to know this.
+
+Now that we know this, we can get on with clearing the window.
+
+Add the following using directive to the top of your file:
+
+```cs
+using System.Drawing;
+```
+
+Then, in your `OnLoad` method, add the following:
+
+```cs
+_gl.ClearColor(Color.CornflowerBlue);
+```
+
+You may notice that this function contains various **overloads**, including ones for a Vector4, and 4 floats. We'll be using the overload that takes `System.Drawing.Color`, because it is the easiest to use and understand.
+
+If you run your application now, you'll notice that the window is still black. That's because we've set the clear *color*, but not actually told OpenGL to clear the window.
+
+In your `OnRender` method, add the following:
+
+```cs
+_gl.Clear(ClearBufferMask.ColorBufferBit);
+```
+
+Run your application again, and you should see a lovely sky blue window!
+
+![Sky blue window](../../../images/opengl/chapter1/cornflower-window.png)
+
+Congrats! You've done your first thing in OpenGL! Didn't work? Check the [source code](../sources/1.2.2-clear-window.html) for this section here.
+
+But - how does OpenGL know to color the window blue? Sure, we've told it we want blue - but we only did that once... How does it know to keep using blue?
+
+Well, this is because OpenGL is a **state machine**.
+
+## OpenGL: The state machine
+Before we can continue further, you'll have to get used to the concept that OpenGL is a **state machine**. It's really hard to understand OpenGL if you don't understand this, so read closely.
+
+### What is a state machine?
+At its core, a state machine holds... state. You set the state of something, and it retains that state until you change it.
+
+This is exactly how OpenGL works. Once you set something, it will remain set until you change it. This counts for everything in OpenGL. Clear color, binding objects, etc etc, everything goes through the state machine. You can *manipulate* the current state, however you have to be wary at all times of what part of the state you are changing. Change the wrong thing and suddenly you program might not work!
+
+So, this explains why clearing the window works. You set the clear color, and it remains *as* the clear color until you change it.
+
+Now that we've got that, let's move on!
+
+## Vertex Array Objects (VAOs)
+A unique feature that modern OpenGL has is what is known as a Vertex Array Object, or VAO. It stores the state of various parameters, such as which buffers you wish to use, and the shader inputs.
```

Comment:
```suggestion
A unique feature that modern OpenGL has is what is known as a Vertex Array Object, or VAO. It stores all of the state needed to supply vertex data, such as which buffers you wish to use and how the vertex data is read from said buffers.
```
I think this will be clearer, what do you think?

From [the khronos docs](https://www.khronos.org/opengl/wiki/Vertex_Specification):
> A Vertex Array Object (VAO) is an [OpenGL Object](https://www.khronos.org/opengl/wiki/OpenGL_Object) that stores all of the state needed to supply vertex data

Reply:
It is clearer, my only concern is that it may confuse some people who don't understand what vertices are, since that hasn't been covered at this part of the tutorial - I'll include something along these lines but I'll try to make it easier to understand for newbies, perhaps something like `It stores all the necessary information required to draw an object to the screen, such as the vertex data (which we'll get into in a minute), and also information on how to process this data.`. How does that sound?

Reply:
Not sure about the part on "how to process this data". All a VAO stores is the "how to read the vertex data". How to process is part of the shader program, which VAOs know nothing about.

Reply:
whoops, yes, this has now been fixed.